### PR TITLE
Aggregate Hybrid neartext/nearvector fixes

### PR DESCRIPTION
**Pull request by donomii**
### What's being changed:

Add nearvector/neartext to aggregate hybrid graphql.  Remove some uneeded code

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/traverser/target_vector_param_helper.go [hunk 1]
```diff
@@ -19,13 +19,7 @@ import (
...
 )
 
-type TargetVectorParamHelper struct{}
-
-func NewTargetParamHelper() *TargetVectorParamHelper {
-	return &TargetVectorParamHelper{}
-}
-
-func (t *TargetVectorParamHelper) GetTargetVectorOrDefault(sch schema.Schema, className, targetVector string) (string, error) {
+func GetTargetVectorOrDefault(sch schema.Schema, className, targetVector string) (string, error) {
```
Comment (antas-marcin) [inline usecases/traverser/target_vector_param_helper.go:22]: I would revert this change and use `TargetVectorParamHelper` struct, this way the code is encapsulated in one struct which purpose is to properly parse target vector settings. I'm against decoupling this to "hanging" methods as then when time goes we tend to forget what those do. I'm a fan of building blocks and encapsulation. which can be easily passed from one service to another

File: adapters/handlers/graphql/local/aggregate/hybrid_search.go [hunk 2]
```diff
@@ -96,3 +140,73 @@ func hybridSubSearch(classObject *graphql.Object,
 		},
 	}
 }
+
+func nearTextFields(prefix string) graphql.InputObjectConfigFieldMap {
```
Comment (antas-marcin) [inline adapters/handlers/graphql/local/aggregate/hybrid_search.go:144]: This definition should be removed, we have unified module arguments in `github.com/weaviate/weaviate/usecases/modulecomponents` package, you can use this package to get the argument definition like this:
```
nearText.New(nearTextTransformer).Arguments()[nearText.NAME]
```
and this part should be added dynamically by `module provider` but this can be fixed later

File: usecases/traverser/target_vector_param_helper.go (hunk 1 above)
Comment (donomii) [reply usecases/traverser/target_vector_param_helper.go:22]: The code isn't encapsulated, it's another file in the traverser namespace.  This is the opposite of encapsulated, it is increasing the already large number of files we have in there.  I will be happy to move it to a helper package.

> I'm against decoupling this to "hanging" methods as then when time goes we tend to forget what those do.

I cannot understand this at all.

> I'm a fan of building blocks and encapsulation. which can be easily passed from one service to another

So am I.  There is no encapsulated data here, which is already a red flag.  What are you encapsulating?

The go standard practice for building blocks and encapsulation is to put the encapsulated code in a package, and then import it.  This provides more namespace protection than mixing a helper class into an important class.  Mixing classes together is the opposite of encapsulation.

In addition, you are reducing code safety and security by adding a nil pointer to a critical data structure.  We have all agreed that we need to improve code quality and program stability, so I would like to know why you are insisting on keeping unsafe code that can panic and potentially corrupt data.

**Review by parkerduckworth — CHANGES_REQUESTED** [review]
Can you please add some acceptance tests?

File: adapters/handlers/graphql/local/aggregate/hybrid_search.go (hunk 2 above)
Comment (antas-marcin) [reply adapters/handlers/graphql/local/aggregate/hybrid_search.go:144]: let's do this change in the next PR

**Comment by donomii** [comment]
> Can you please add some acceptance tests?

Tests are on the hybrid_groupby branch

**Review by tsmith023 — APPROVED** [review]
